php-checks/src/main/java/org/sonar/php/checks/ConstructWithParenthesesCheck.java

```diff
@@ -0,0 +1,132 @@
+
+@Rule(key = ConstructWithParenthesesCheck.KEY)
+public class ConstructWithParenthesesCheck extends PHPVisitorCheck {
+
+  public static final String KEY = "S6600";
+  private static final String MESSAGE = "Remove the parentheses from this \"%s\" call.";
+  private static final Set<String> CONSTRUCT_FUNCTION = Set.of("echo", "clone", "include", "include_once", "require", "require_once", "print");
+
+  @Override
+  public void visitCaseClause(CaseClauseTree tree) {
+    raiseIssueOnParenthesizedExpression(tree.expression(), tree.caseToken(), tree.caseSeparatorToken(), "case");
+    super.visitCaseClause(tree);
+  }
+
+  @Override
+  public void visitBreakStatement(BreakStatementTree tree) {
+    raiseIssueOnParenthesizedExpression(tree.argument(), tree, "break");
+    super.visitBreakStatement(tree);
+  }
+
+  @Override
+  public void visitReturnStatement(ReturnStatementTree tree) {
+    raiseIssueOnParenthesizedExpression(tree.expression(), tree, "return");
+    super.visitReturnStatement(tree);
+  }
+
+  @Override
+  public void visitThrowStatement(ThrowStatementTree tree) {
+    raiseIssueOnParenthesizedExpression(tree.expression(), tree, "throw");
+    super.visitThrowStatement(tree);
+  }
+
+  @Override
+  public void visitYieldExpression(YieldExpressionTree tree) {
+    String constructName = tree.fromToken() != null ? "yield from" : "yield";
+    ExpressionTree elementToCheck = tree.key() != null ? tree.key() : tree.value();
+    raiseIssueOnParenthesizedExpression(elementToCheck, tree, constructName);
+    super.visitYieldExpression(tree);
+  }
+
+  @Override
+  public void visitContinueStatement(ContinueStatementTree tree) {
+    raiseIssueOnParenthesizedExpression(tree.argument(), tree, "continue");
+    super.visitContinueStatement(tree);
+  }
+
+  private void raiseIssueOnParenthesizedExpression(@Nullable ExpressionTree expression, Tree raiseIssueOn, String constructName) {
+    if (isParenthesizedExpression(expression)) {
+      newIssue(raiseIssueOn, String.format(MESSAGE, constructName));
+    }
+  }
+
+  private void raiseIssueOnParenthesizedExpression(@Nullable ExpressionTree expression, Tree raiseIssueOnStart, Tree raiseIssueOnEnd, String constructName) {
+    if (isParenthesizedExpression(expression)) {
+      context().newIssue(this, raiseIssueOnStart, raiseIssueOnEnd, String.format(MESSAGE, constructName));
+    }
+  }
+
+  private static boolean isParenthesizedExpression(@Nullable ExpressionTree expression) {
+    return expression != null && expression.is(Tree.Kind.PARENTHESISED_EXPRESSION);
+  }
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    String constructName = getConstructName(tree.callee());
+    if (isConstructFunction(constructName) && isParenthesized(tree)) {
+      newIssue(tree, String.format(MESSAGE, constructName));
+    }
+    super.visitFunctionCall(tree);
+  }
+
+  private static boolean isParenthesized(FunctionCallTree tree) {
```

Comment:
Same here. `isParenthesized` and a function call as an argument may lead to the idea that you are looking for something like `(echo FOO)`. My suggestion is `hasParenthesizedArgument`.